conan/tools/cmake/toolchain/blocks.py

```diff
@@ -502,6 +505,38 @@ class FindFiles(Block):
         {% endif %}
     """)
 
+    @staticmethod
+    def _multiconfig_generator(prop):
+        return ' '.join(f'"$<$<CONFIG:{c}>:{i}>"' for c, v in prop.items() for i in v)
+
+    def _get_host_runtime_dirs_mc(self, host_req):
+        settings = self._conanfile.settings
+        host_runtime_dirs = {}
+        is_win = self._conanfile.settings.get_safe("os") == "Windows"
+
+        # Get the previous configuration
+        if os.path.exists(CONAN_TOOLCHAIN_FILENAME):
```

Comment:
this should also check for `is_multi_configuration(self._toolchain.generator)`, otherwise if we are in single configuration land, it can mix up the output from different invocations of `conan install` (it could be the same config, but with a graph with previous versions potentially)